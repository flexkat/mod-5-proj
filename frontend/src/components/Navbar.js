import React from 'react';
import { NavLink } from 'react-router-dom'
import { Menu } from 'semantic-ui-react'


class Navbar extends React.Component {

  state = {}

  handleItemClick = (e, { name }) => this.setState({ activeItem: name })

  render() {
    const { activeItem } = this.state
    return (
      <Menu pointing>
        <Menu.Item as={NavLink} to="/" exact strict name="home" active={activeItem === "home"} onClick={this.handleItemClick}>Home</Menu.Item>
        <Menu.Item as={NavLink} to="/setup" exact name="setup" active={activeItem === "setup"} onClick={this.handleItemClick}>Add medicine</Menu.Item>
        <Menu.Item as={NavLink} to="/statistics" exact name="stats" active={activeItem === "stats"} onClick={this.handleItemClick}>Statistics</Menu.Item>
        <Menu.Item as={NavLink} to="/contact" exact name="contact" active={activeItem === "contact"} onClick={this.handleItemClick}>Contact</Menu.Item>
        {/* <Menu.Item header position="right">Pill Pal</Menu.Item> */}
      </Menu>
    );
  }
}

export default Navbar;