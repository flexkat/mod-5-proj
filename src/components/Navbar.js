import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router'

class Navbar extends React.Component {
  render() {
    return (
      <Router >
      <div className="Navbar">
        <Route path="/" exact>Home</Route>
      </div>
      </Router >
    );
  }
}

export default Navbar;