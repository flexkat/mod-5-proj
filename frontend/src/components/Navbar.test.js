import React from 'react';
import Navbar from './Navbar'
import { shallow } from 'enzyme';


describe("nav bar component", () => {
  it("should render the menu when no props are passed through", () => {
    const wrapper = shallow(<Navbar />);

    expect(wrapper).toMatchSnapshot();
  })
})