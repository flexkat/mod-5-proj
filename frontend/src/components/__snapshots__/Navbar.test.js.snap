// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav bar component should render the menu when no props are passed through 1`] = `
<Menu
  pointing={true}
>
  <MenuItem
    active={false}
    as={[Function]}
    exact={true}
    name="home"
    onClick={[Function]}
    strict={true}
    to="/"
  >
    Home
  </MenuItem>
  <MenuItem
    active={false}
    as={[Function]}
    exact={true}
    name="setup"
    onClick={[Function]}
    to="/setup"
  >
    Add my medicine
  </MenuItem>
  <MenuItem
    active={false}
    as={[Function]}
    exact={true}
    name="stats"
    onClick={[Function]}
    to="/statistics"
  >
    Statistics
  </MenuItem>
  <MenuItem
    position="right"
  >
    Pill Pal
  </MenuItem>
</Menu>
`;
